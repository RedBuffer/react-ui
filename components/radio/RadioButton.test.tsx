import * as React from 'react';
import { mount } from 'enzyme';

import RadioButton from './RadioButton';

describe('Radio Button', () => {
  let handleChange;
  let wrapper;

  beforeEach(() => {
    handleChange = jest.fn((e) => e.target.value);
    wrapper = mount(
      <RadioButton name="radio-test" id={'radio'} onChange={handleChange}>
        radio
      </RadioButton>,
    );
  });

  it('should render correctly', () => {
    const component = wrapper;

    expect(component).toMatchSnapshot();
  });

  it('click on radio button', () => {
    wrapper
      .find('#radio')
      .hostNodes()
      .simulate('change', { target: { checked: true } });

    expect(handleChange.mock.calls.length).toEqual(1);
    expect(handleChange.mock.calls[0][0].target.checked).toEqual(true);
  });

  it('click on disabled checkbox', () => {
    wrapper.setProps({ disabled: true });
    wrapper.find('#radio').hostNodes().simulate('click');

    expect(handleChange.mock.calls.length).toEqual(0);
  });
});
