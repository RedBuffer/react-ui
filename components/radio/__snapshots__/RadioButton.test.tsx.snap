// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button should render correctly with menu 1`] = `
<Component
  id="radio"
  name="radio-test"
  onChange={[MockFunction]}
>
  <label
    className="m-2 block relative select-none text-md cursor-pointer pl-8 react-ui-radio-label"
  >
    radio
    <input
      className="h-0 w-0 absolute opacity-0 react-ui-radio"
      disabled={false}
      id="radio"
      name="radio-test"
      onChange={[MockFunction]}
      type="radio"
    />
    <span
      className="top-0 left-0 absolute bg-gray-400 rounded-full h-6 w-6 p-2 checkmark-md react-ui-radio-checkmark"
    />
  </label>
</Component>
`;
